var/spack/repos/builtin/packages/py-amrex/package.py

```diff
@@ -63,46 +72,46 @@ class PyAmrex(PythonPackage, CudaPackage, ROCmPackage):
         depends_on("amrex dimensions=3")
     with when("+mpi"):
         depends_on("amrex +mpi")
+    with when("~mpi"):
+        depends_on("amrex ~mpi")
     with when("+openmp"):
         depends_on("amrex +openmp")
+    with when("~openmp"):
+        depends_on("amrex ~openmp")
     with when("+tiny_profile"):
         depends_on("amrex +tiny_profile")
     with when("+cuda"):
         depends_on("amrex +cuda")
         # todo: how to forward cuda_arch?
+    with when("~cuda"):
+        depends_on("amrex ~cuda")
     with when("+rocm"):
         depends_on("amrex +rocm")
         # todo: how to forward amdgpu_target?
+    with when("~rocm"):
+        depends_on("amrex ~rocm")
+    with when("+sycl"):
+        depends_on("amrex +sycl")
+    with when("~sycl"):
+        depends_on("amrex ~sycl")
 
     depends_on("py-pytest", type="test")
     depends_on("py-pandas", type="test")
     depends_on("py-cupy", type="test", when="+cuda")
 
     tests_src_dir = "tests/"
+    install_targets = ["install", "pip_install"]
```

Comment:
This seems to install into the site packages of the `python-venv` install location, not the `py-amrex` install location. It works, but is that what we want? Otherwise, I can set more flags if we want this under the `py-amrex` install prefix.

Reply:
@adamjstewart do you know what I need to do to a `pip install` command to ensure it's local and not in the global venv?

Reply:
You need to specify the `--prefix`. Look at the `PythonPackage` base class for details. Also make sure to use `std_pip_args` and extend the `PythonExtension` base class.

Reply:
Thank you so much, Adam  -- added now :)